kafka-rest/src/main/java/io/confluent/kafkarest/response/StreamingResponse.java

```diff
@@ -246,13 +284,17 @@ private void push(CompletableFuture<ResultOrError> result) {
                       ResultOrError res = result.join();
                       log.debug("Writing to sink");
                       sink.write(res);
-                      if (res instanceof ErrorHolder
-                          && ((ErrorHolder) res).getError().getErrorCode() == 429) {
-                        log.warn(
-                            "Connection closed as a result of rate limiting being exceeded "
-                                + "for longer than grace period.");
-                        sinkClosed = true;
-                        sink.close();
+                      if (res instanceof ErrorHolder) {
+                        if (res.asError().getError().getErrorCode() == 429) {
```

Comment:
nit: Why not just start with `Error error = ((ErrorHolder) res).getError()` and avoid the redundant `asError()` method?